sympy/solvers/ode/tests/test_riccati.py

```diff
@@ -0,0 +1,903 @@
+
+
+def test_val_at_inf():
+    tests = [
+    (
+        Poly(12*x**8 - 12*x**7 - 11*x**6 + 8*x**5 + 3*x**4 - x**3 + x**2 - 11*x, x),
+        Poly(-14*x**2 + x, x),
+         -6
+    ),
+    (
+        Poly(-8*x**4 + 14*x**3 + 10*x**2 + 3*x - 11, x),
+        Poly(-9*x**4 + 3*x**3 + 15*x**2 - 6*x - 14, x),
+         0
+    ),
+    (
+        Poly(-6*x**3 - 8*x**2 + 8*x - 6, x),
+        Poly(-5*x**3 + 12*x**2 - 6*x - 9, x),
+         0
+    ),
+    (
+        Poly(15*x**3 - 5*x**2 - 6*x - 2, x),
+        Poly(9*x**5 + 12*x**4 + 7*x**3 + 11*x**2 - 2*x - 9, x),
+         2
+    ),
+    (
+        Poly(10*x**3 + 8*x**2 - 13*x + 6, x),
+        Poly(-13*x**10 - x**9 + 5*x**8 + 7*x**7 + 10*x**6 + 6*x**5 - 7*x**4 + 11*x**3 - 8*x**2 + 5*x + 13, x),
+         7
+    ),
+    (
+        Poly(-15*x - 15, x),
+        Poly(15*x**3 - 13*x**2 - 4*x + 2, x),
+         2
+    ),
+    (
+        Poly(-9*x**8 - 9*x**7 + 11*x**6 - 11*x**5 + 9*x**4 + 7*x**3 - 8*x**2 - 5*x - 12, x),
+        Poly(9*x**10 + 4*x**9 + 12*x**8 - 8*x**7 - 3*x**6 - 14*x**5 + 8*x**4 - 15*x**3 + 4*x**2 + 9*x + 1, x),
+         2
+    ),
+    (
+        Poly(13*x - 9, x),
+        Poly(-8*x**4 + 5*x**3 - 4*x**2 - 14*x + 4, x),
+         3
+    ),
+    (
+        Poly(12*x**7 - 7*x**6 + 9*x**5 + 6*x**4 - 14*x**3 + 15*x**2 - x + 8, x),
+        Poly(11*x**10 + 2*x**9 + x**8 - 3*x**7 - 12*x**6 - 10*x**5 - 15*x**4 - 14*x**3 - 5*x**2 + 15*x + 15, x),
+         3
+    ),
+    (
+        Poly(5*x**6 + 9*x**5 - 11*x**4 - 9*x**3 + x**2 - 4*x + 4, x),
+        Poly(15*x**4 + 3*x**3 - 8*x**2 + 15*x + 12, x),
+         -2
+    )]
+    for num, den, val in tests:
+        assert val_at_inf(num, den, x) == val
+
+
+def test_necessary_conds():
+    # Valuation at Infinity is an odd negative integer
+    assert check_necessary_conds(-3, [1, 2, 4]) == False
+    # Valuation at Infinity is a positive integer lesser than 2
+    assert check_necessary_conds(1, [1, 2, 4]) == False
+    # Multiplicity of a pole is an odd integer greater than 1
+    assert check_necessary_conds(2, [3, 1, 6]) == False
+    # All values are correct
+    assert check_necessary_conds(-10, [1, 2, 8, 12]) == True
+
+
+def test_inverse_transform_poly():
+    fns = [
+    (30*x**6 - 12*x**5 + 15*x**4 - 15*x**2 + 10*x + 60)/(3*x**10 - 45*x**9 + 15*x**5 + 15*x**4 - 5*x**3 \
+    + 15*x**2 + 45*x - 15),
+
+    (15*x**3 - 8*x**2 - 2*x - 6)/(18*x + 6),
+
+    (-75*x**6 + 12*x**5 - 80*x**4 + 15*x**3 - 75*x**2 + 120*x + 120)/(80*x**9 - 48*x**8 - 40*x**6 - \
+    80*x**5 - 100*x**4 - 60*x**2 + 60*x + 60),
+
+    (-40*x**4 + 180*x**3 - 60*x**2 + 60*x + 150)/(30*x**6 - 40*x**5 - 20*x**4 - 120*x**3 - 45*x - 48),
+    (30*x**9 - 60*x**8 - 40*x**7 - 60*x**6 + 15*x**5 - 240*x**4 + 300*x**3 + 180*x**2 + 30*x - 90)/( \
+    60*x**10 - 120*x**9 - 60*x**8 - 240*x**7 - 48*x**6 + 60*x**5 + 30*x**4 + 150*x**3 + 30*x**2 - 60*x),
+    (24*x**10 + 20*x**7 - 48*x**5 - 45*x**4 - 60*x**3 + 60*x**2 + 60*x + 20)/(30*x**5 - 45*x**4 - 48*x**3
+    - 24*x**2 + 30*x),
```

Comment:
Do these examples need to be so long? Why not an example that is short enough that you can easily see that it is correct?

There is nothing wrong with using randomised testing but we don't want to explicitly include large, complicated randomly generated examples in the tests unless they represent some particular corner case that can not be reproduced with a simpler example.

Reply:
I'll remove this test case and make others simpler.